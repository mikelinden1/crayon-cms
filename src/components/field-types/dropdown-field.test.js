import DropdownField from './dropdown-field';

const props = {
    name: 'test',
    value: 'val1',
    options: [
        { label: 'val1', value: 'val1'},
        { label: 'val2', value: 'val2'},
        { label: 'val3', value: 'val3'}
    ],
    onChange: () => {},
    disabled: false,
    actions: {
        fetchDatasource: () => {}
    }
};

it('DropdownField matches snapshot', () => {
    const wrapper = shallow(<DropdownField {...props} />);
    expect(wrapper).toMatchSnapshot();
});
