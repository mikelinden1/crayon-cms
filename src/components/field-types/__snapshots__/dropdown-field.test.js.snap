// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownField matches snapshot 1`] = `
<Select
  arrowRenderer={[Function]}
  autosize={true}
  backspaceRemoves={true}
  backspaceToRemoveMessage="Press backspace to remove {label}"
  clearAllText="Clear all"
  clearRenderer={[Function]}
  clearValueText="Clear value"
  clearable={true}
  closeOnSelect={true}
  deleteRemoves={true}
  delimiter=","
  disabled={false}
  escapeClearsValue={true}
  filterOptions={[Function]}
  ignoreAccents={true}
  ignoreCase={true}
  inputProps={Object {}}
  isLoading={false}
  joinValues={false}
  labelKey="label"
  matchPos="any"
  matchProp="any"
  menuBuffer={0}
  menuRenderer={[Function]}
  multi={false}
  name="test"
  noResultsText="No results found"
  onBlurResetsInput={true}
  onChange={[Function]}
  onCloseResetsInput={true}
  onSelectResetsInput={true}
  openOnClick={true}
  optionComponent={[Function]}
  options={
    Array [
      Object {
        "label": "val1",
        "value": "val1",
      },
      Object {
        "label": "val2",
        "value": "val2",
      },
      Object {
        "label": "val3",
        "value": "val3",
      },
    ]
  }
  pageSize={5}
  placeholder="Select..."
  removeSelected={true}
  required={false}
  rtl={false}
  scrollMenuIntoView={true}
  searchable={true}
  simpleValue={false}
  tabSelectsValue={true}
  trimFilter={true}
  value={
    Object {
      "label": "val1",
      "value": "val1",
    }
  }
  valueComponent={[Function]}
  valueKey="value"
/>
`;
